### ISSUE TITLE
Allow peacock settings to be saved and retrieved from file other than `.vscode/settings.json`

### ISSUE THREAD

**Issue by marvingreenberg** [johnpapa/vscode-peacock#512]
**Describe the solution you'd like**
I'd like the ability to set a global setting like 
```
  "peacock.configFile": "${workspaceFolder}/.vscode/peacock-settings.json"
```
And then when peacock saves and retrieves colors for a project from a separate file, than the workspace `.vscode/settings.json`.  In certain projects I have found it useful to commit `.vscode/settings.json` to version control - say when defining debugging configurations or other properties that SHOULD be committed, to help collaborate with other using vscode.  But the color scheme for a workspace is a personal preference.

**Describe alternatives you've considered**
None really.  It may be that moving settings into a separate file is complex and violates principals of the vscode extensions architecture.  Simplest may be to just say "won't fix", and its a minor point.

**Comment by johnpapa** [comment]
Thank you @marvingreenberg for creating this issue!

**Comment by NekoApocalypse** [comment]
I second this. In our project we want to sync certain linter and debug settings so we do commit `.vscode/settings.json` to version control.

And it seems like there is no elegant way for an extension to read settings from places that are not handled by vscode itself.

Perhaps setting up a [multi-root workspace](https://code.visualstudio.com/docs/editor/multi-root-workspaces) would help. But I haven't tried it yet.

**Comment by royemosby** [comment]
bump. I Love Peacock but I don't like `.vscode` directories littering my filesystem when I update with my computer's configs in VS Code. It would be nice if the plugin would save all themes in a single config.

**Comment by MartinSGill** [comment]
Themes don't write themselves into `settings.json`, but instead into user settings: `%APPDATA%\Code\User\settings.json`

Perhaps that's where Peacock needs to write its settings as well. It would need to be able to save settings based on workspaces so possibly a structure like this:

```
  "peackock": {
    "workspaces": {
      "<workspacename/path>": {
        "activityBar.activeBackground": "#ff6433",
        "...": "...",
        "titleBar.inactiveBackground": "#ff3d0099",
        "titleBar.inactiveForeground": "#e7e7e799"
      }
    }
  }
```

**Comment by MartinSGill** [comment]
Having looked around a bit, isn't this a duplicate of #7 ?

**Comment by andrzejwalk** [comment]
has anyone found a workaround?

**Comment by DallasO** [comment]
_TL;DR - A workaround is, don't commit the Peacock configuration?_

In my case at least, we rarely update `settings.json`. We set our team configuration at the beginning of the project and have not committed a change since.
I'm the only one on my team using Peacock as far as I'm aware, and rarely stage changes using `git add -all`, but there have been cases I accidentally committed and pushed my Peacock configuration.
The last change we needed, I stashed my Peacock settings, made the required change, and unstashed.

I'm more annoyed at seeing a pending change all the time in the Source Control tab, and having to remove Peacock's settings to remove work trees (I extensively use `git worktree`).

Some other solutions such as #7 or #517 defeat the purpose of this extension for me, and would be detrimental with work trees. How would I easily know I'm working on our production branch vs a feature branch?

Having a dedicated settings file for this extension would be ideal, but I assume it is not possible to override the VSCode theme in such a manner.

**Comment by AlvesJorge** [comment]
Bump. This is still needed

**Comment by joeytwiddle** [comment]
In the meantime, I recommend using `git add -p` to stage the changes you do want, and not those you don't.

Hit `n` when it offers to stash the Peacock settings, which you don't want to share.

Hit `s` if it offers you one big hunk, to split them up. This requires at least one line separation between your new settings and the Peacock settings.

**Comment by johnpapa** [comment]
Closing — `workbench.colorCustomizations` must live in `settings.json` or `.code-workspace` — that's where VS Code reads color overrides from. No alternative file path is supported by VS Code's theming engine. The real solution to keeping personal colors out of shared files is #495 (user-level workspace→color mappings). Follow #495 for progress. Thanks! 🙏
